### ISSUE TITLE
Improve Boost.Test Version Identification

### ISSUE THREAD

**Issue by netspiri** [etas/vs-boost-unit-test-adapter#173]
Boost 1.63 has support for the new command-line option `--version` which launches the test module, displays version information and exits. The output is written to standard error in a format as follows:

```
Boost.Test module in executable 'test.exe'
Compiled from Boost version 1.63.0 with static linking to Boost.Test
- Compiler: Microsoft Visual C++ version 14.0
- Platform: Win32
- STL     : Dinkumware standard library version 650
```

Make use of this information to better react to version differences in Boost Test.

Hint(s):
- Given that Boost 1.61 and 1.62 are to be supported, make use of debug symbols (similar to `--list_content`) as a primary step to identify versions.
- Execute the process one additional time during discovery using the `--version` argument to extract the necessary information. This information can then be serialized and passed onto the executor via the `Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase.SetPropertyValue` and `Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase.GetPropertyValue` methods.

**Comment by guwirth** [comment]
@netspiri would still prefer to have the version number in the output/report files instead of starting the process an additional time.

**Comment by netspiri** [comment]
We can extract version information from the output/report files if we run tests using the `--build_info` argument. Additional metadata will be added to the test report.

Alas we need version information as soon as possible i.e. before test discovery and not after test execution. Unfortunately, we cannot use the same approach with `--list_content=DOT`. Running `--build_info --list_content=DOT` will only provide test discovery information.

I do not think that running the test module an additional time will cause a big impact in terms of performance. Note that the process will only run 1 additional time, for example, having a test module with 10 tests, the process will be launched once to identify the version, another to run `--list_content` and up to 10 additional times to run each and every test based on user selection and test batch strategy, leading to a maximum of 12 process launches in the worst case scenario.

In conclusion, I think this is the best option that we currently have and it is best that we utilise the available mechanism to avoid versioning issues in the future.
